application-portal/src/app/services/choices/choices.service.ts

```diff
@@ -1,9 +1,284 @@
+  design: Array<string> = ["Digital Design Course"];
+
+
+
+  constructor() { 
+    // Constructor populates all arrays, TESTED
+
+    // Get programs
+    fetch('https://students-staging.ixperience.co.za/api/v1/programs.json')
+    .then(response => {
+      return response.json();
+    })
+    .then(data => {
+      this.programs = data;
+      console.log(this.programs);
+    })
+    .catch(err => {
+      console.log(err);
+    })
+
+    // Get locations
+    fetch('https://students-staging.ixperience.co.za/api/v1/locations.json')
+    .then(response => {
+      return response.json();
+    })
+    .then(data => {
+      this.locations = data;
+      console.log(this.locations);
+    })
+    .catch(err => {
+      console.log(err);
+    })
+
+    // Get periods
+    fetch('https://students-staging.ixperience.co.za/api/v1/periods.json')
+    .then(response => {
+      return response.json();
+    })
+    .then(data => {
+      this.periods = data;
+      console.log(this.periods);
+    })
+    .catch(err => {
+      console.log(err);
+    })
+
+    // Get sessions
+    fetch('https://students-staging.ixperience.co.za/api/v1/sessions.json')
+    .then(response => {
+      return response.json();
+    })
+    .then(data => {
+      this.sessions = data;
+      console.log(this.sessions);
+    })
+    .catch(err => {
+      console.log(err);
+    })
+
+    // TODO: get current user from user service
+  }
+
+  /* 
+  TODO Mockups of functions by page
+    - Fields -> Courses:
+      - getProgramsByField(field: string) <-- inputs are tech, business, design
+        - search programs array and returns array of programs with that field
+        - create array for each field with the program names in the array for this
+    - Courses -> Course Info: 
+      - getDatesByProgram(program_id: number)
+        - search sessions array for sessions with that program, 
+          then for each session, get period from period_id (separate function? probably) 
+          and return array with period objects
+        - TODO: break this up into multiple functions? it's quite complicated
+      - (NOT IN MOCKUP BUT) getDescriptionByCourseId? no place for it tho but we have the info...
+    - Course Info -> Cities:
+      - getCitiesByProgram(program: string)
+        - searches sessions array for sessions with that program,
+          then for each session, get city from city_id and return array with city objects
+    - Cities -> City Info:
+      - ???? Would this be the description? In which case just getCity would work?
+  */
+
+  /* Basic Get/Set for all of the fields in Choices, UNTESTED */
+
+  getProgram() {
+    return this.program;
+  }
+
+  setProgram(program_id: number) {
+    for (var i = 0; i < this.programs.length; i++) {
+      if (this.programs[i].id == program_id) {
+        // set choices' program id
+        this.choices.program_id = this.programs[i].id;
+        // set program variable 
+        this.program = this.programs[i];
```

Comment:
`Array.find` is a cleaner way to search an array for something (without having to loop through it yourself)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find